src/stash/src/lib.rs

```diff
@@ -860,7 +860,7 @@ where
     /// Iterates over the items viewable in the current transaction in arbitrary
     /// order.
     pub fn for_values<F: FnMut(&K, &V)>(&self, mut f: F) {
-        let mut seen = HashSet::with_capacity(self.pending.len());
+        let mut seen = BTreeSet::new();
```

Comment:
I'll leave this (and all the stash changes) for @mjibson to comment on. He's been working on stash performance and would understand the performance implications better than me.

Reply:
See my other comment: I think we should focus on preventing iteration over hashmaps, not the use in general.